boa_parser/src/parser/expression/primary/async_function_expression/mod.rs

```diff
@@ -124,6 +113,26 @@ where
             )));
         }
 
+        // Early Error: If BindingIdentifier is present and the source code matching BindingIdentifier is strict mode code,
+        // it is a Syntax Error if the StringValue of BindingIdentifier is "eval" or "arguments".
+        if (cursor.strict_mode() || body.strict())
+            && [Some(Sym::EVAL), Some(Sym::ARGUMENTS)].contains(&name.map(Identifier::sym))
+        {
+            return Err(Error::lex(LexError::Syntax(
+                "unexpected identifier 'eval' or 'arguments' in strict mode".into(),
+                name_span.start(),
+            )));
+        }
```

Comment:
Would be better to use `if let Some(name) = name` to save us the check in case the name is `None`, and to simplify the logic of the `contains` call. Also applies to the other functions' parsing.

Reply:
I think it's worth. And moreso knowing `if let` chains are coming soon™.